### PR TITLE
Move 'buildSrc' to 'build-logic' and introduce 'build-logic-commons'

### PR DESCRIPTION
**Pull request by jjohannes**
- Turn `buildSrc` into included `build-logic` build
- Introduce `build-logic-commons` for
   - What was previously in the root build script of `buildSrc`
   - What was previously in `config` and `gradle/shared-with-buildSrc/code-quality-configuration.gradle.kts` (code quality checks shared by main and build logic builds)

### CODE REVIEW

**Comment by jjohannes** [comment]
@bot-gradle test this

**Comment by jjohannes** [comment]
@bot-gradle test this

**Comment by jjohannes** [comment]
@bot-gradle test this

**Comment by jjohannes** [comment]
@bot-gradle test this

**Comment by jjohannes** [comment]
@bot-gradle test this

**Comment by jjohannes** [comment]
@bot-gradle test this

File: build-logic-commons/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+tasks.register("check") {
+    dependsOn(subprojects.map { "${it.name}:check" })
+}
```
Comment (JLLeitschuh) [inline build-logic-commons/build.gradle.kts:3]: Still annoying that Gradle doesn't do this automatically.

Can you do the same thing for `test` and `sanityCheck`?

File: build-logic-commons/code-quality/src/main/kotlin/gradlebuild.code-quality.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+}
+
+dependencies {
+    rules("gradlebuild:code-quality-rules") {
+        because("Provides rules defined in XML files")
+    }
+    codenarc("gradlebuild:code-quality-rules") {
+        because("Provides the IntegrationTestFixturesRule implementation")
+    }
+    codenarc("org.codenarc:CodeNarc:1.5")
```
Comment (JLLeitschuh) [inline build-logic-commons/code-quality/src/main/kotlin/gradlebuild.code-quality.gradle.kts:60]: Should this version be extracted?

Comment (jjohannes) [reply build-logic-commons/code-quality/src/main/kotlin/gradlebuild.code-quality.gradle.kts:60]: With extraction you mean defined in a central place with the other library versions?

For that we need to pull out the "platforms" (potentially catalogs in the future) into a separate build which we can then use in all other included builds. We should do that at some point but there are some point. But I don't want to mix this into this PR. And there is only this one dependency in 'build-logic-commons' at the moment.

File: build-logic-commons/build.gradle.kts (hunk 1 above)
Comment (jjohannes) [reply build-logic-commons/build.gradle.kts:3]: This is added here and wired into `sanityCheck`. I think that is all we need for starters.

So now if you or CI runs `sanityCheck` all build logic checks will execute as part of that.

If you want to work with this build on your local machine with individual tasks, you can also `cd <included-build>`. And then run tasks in that build individually.

**Comment by jjohannes** [comment]
@bot-gradle test this

**Pull request merged**